serialization/tests.py

```diff
@@ -0,0 +1,196 @@
+from homework import *
+import unittest
+
+
+class TestCar(unittest.TestCase):
+
+    def setUp(self):
+        self.car1 = Car(price=1000, type="Diesel", producer="BMW", mileage=10000, number=1234)
+        self.car2 = Car(price=1000, type="Diesel", producer="BMW", mileage=10000, number=1234)
+        self.car3 = Car(2000, "Van", "Chery", 20000)
+
+    def test_car_change_number_success(self):
+        self.car1.change_number('c857fd34-b3e8-476a-833a-e38e1f928f6e')
+        self.assertIsInstance(self.car1.number, uuid.UUID)
+
+    def test_car_change_number_fail(self):
+        self.car1.change_number(12345)
+        self.assertNotIsInstance(self.car1.number, uuid.UUID)
+
+    def test_convert_car_to_dict_success(self):
+        self.assertIsInstance(self.car1.obj_to_dict(), dict)
+
+    def test_car_equality_success(self):
+        self.assertTrue(self.car1.car_equality(self.car2))
+
+    def test_car_equality_fail(self):
+        self.assertFalse(self.car1.car_equality(self.car3))
+
+    def test_convert_car_to_dict_and_back_success(self):
+        car3_to_dict = json.dumps(self.car3.obj_to_dict(), cls=JsonEncoder, indent=4)
+        car3_to_obj = json.loads(car3_to_dict, object_hook=dict_to_obj)
+        self.assertTrue(self.car3.car_equality(car3_to_obj))
+
+    def test_magic_method__str__for_car_success(self):
+        expected_result = f"""
+        Attributes of current Car:
+            price: '2000.0';
+            type: 'Van';
+            producer: 'Chery';
+            number: '{self.car3.number}';
+            mileage: '20000.0';
+        """
+        self.assertEqual(str(self.car3), expected_result)
+
+    def test_magic_method__repr__for_car_success(self):
+        expected_result = "Car(price='2000.0', type='Van', producer='Chery', mileage='20000.0')"
+        self.assertEqual(repr(self.car3), expected_result)
+
+
+class TestGarage(unittest.TestCase):
+
+    def setUp(self):
+        self.car1 = Car(1000, "Diesel", "BMW", 10000)
+        self.car2 = Car(1000, "Diesel", "BMW", 10000, 'c857fd34-b3e8-476a-833a-e38e1f928f6e')
+        self.car3 = Car(2000, "Van", "Chery", 20000, 1111)
+        self.garage1 = Garage('Kiev', [self.car1], 2)
+        self.garage2 = Garage('Kiev', [self.car1, self.car2], 2)
+        self.garage3 = Garage('Kiev', [self.car1, self.car2], 2)
+
+    def test_add_car_to_garage_success(self):
+        expected_result = """
+        Attributes of current Garage:
+            town: 'Kiev';
+            total cars: '2';
+            places: '2';
+            owner: 'None';
+        """
+        self.garage1.add_car(self.car2)
+        self.assertEqual(str(self.garage1), expected_result)
+
+    def test_add_car_to_garage_fail(self):
+        error_message = self.garage2.add_car(self.car3)
```

Comment:
In success case you return nothing although in fail case you return message. It`s wrong logic